### DISCUSSION TITLE
Where to find list of all possible gridpoints in the US?

### DISCUSSION THREAD

**Discussion by multibdoyle** [weather-gov/api#686] [General]
Hello - I would like to retrieve a list of all gridpoint forecast endpoints in the U.S. (e.g. endpoints in the format https://api.weather.gov/gridpoints/{office}/{gridX},{gridY}/forecast). I have not been able to find this full list of gridpoints anywhere and the method I have of retrieving all endpoints by searching for the gridpoints from an iteration of all latitude / longitudes only seems to give me coverage for about 900 gridpoint pairs. 

Can someone point me in the direction of a list of all gridpoint pairs, OR provide some code outlining how to retrieve all possible gridpoint pairs? Or explain to me where I may be headed in the wrong direction / lacking in understanding about gridpoints :) Thank you!

**Comment by sullynole** [comment]
This does not exist since each WFO defines its own domain that extends beyond its boundaries and all grids are not accessible as only gridpoints within a WFO's boundary will be returned.  The gridpoints endpoint is designed to retrieve a point forecast through the points endpoint, so pulling all gridpoints is not recommended.  See #621 for further discussion on this topic.
